src/main/java/org/spongepowered/common/inject/InjectionPointProvider.java

```diff
@@ -66,20 +73,35 @@ public boolean matches(Binding<?> binding) {
         }
     }
 
+    @Nullable
     private static InjectionPoint findInjectionPoint(List<DependencyAndSource> dependencyChain) {
```

Comment:
It's worth noting that [`ProvisionListener$ProvisionInvocation#getDependencyChain()`](https://github.com/google/guice/blob/master/core/src/com/google/inject/spi/ProvisionListener.java#L66-L102) has been deprecated ([not long after `4.1` was released](https://github.com/google/guice/commit/6f185140d88feafeec65039a213f08a0abeff268)) and scheduled for removal in `4.4`, as is [`DependencyAndSource`](https://github.com/google/guice/blob/master/core/src/com/google/inject/spi/DependencyAndSource.java#L31-L33). 

Would you be able/willing to see about making this use the alternative method (as suggested in the javadocs) now, rather than later, to ensure that what we do is still possible down the road? If not, could you add a comment to this class, please?